connectors/src/connectors/confluence/temporal/workflows.ts

```diff
@@ -40,13 +43,25 @@ export async function confluenceFullSyncWorkflow({
   const spaceIdsToSync =
     spaceIdsToBrowse ?? (await getSpaceIdsToSyncActivity(connectorId));
 
+  const uniqueSpaceIds = new Set(spaceIdsToSync);
+
+  setHandler(spaceUpdatesSignal, (spaceUpdates: SpaceUpdatesSignal[]) => {
+    for (const { action, spaceId } of spaceUpdates) {
+      if (action === "added") {
+        uniqueSpaceIds.add(spaceId);
+      } else {
+        uniqueSpaceIds.delete(spaceId);
+      }
+    }
+  });
+
   const {
     workflowId,
     searchAttributes: parentSearchAttributes,
     memo,
   } = workflowInfo();
 
-  for (const spaceId of spaceIdsToSync) {
+  for (const spaceId of uniqueSpaceIds) {
```

Comment:
Please add a comment here explaning that it actually works.

Reply:
Comment added. I also added an example of a run in the PR's description.